### DISCUSSION TITLE
Is it ok I published CLI11 as a NuGet pkg?

### DISCUSSION THREAD

**Discussion by diogo-strube** [CLIUtils/CLI11#777] [Q&A]
Hi folks! I was using this amazing library and, as my projects heavily used NuGet packages, I just went with the flow (extreme programming mode) and packaged CLI11 as a NuGet Pkg:

<img src="https://user-images.githubusercontent.com/19294348/190648613-6e15ed7f-d8df-495c-bd2d-3b9e994cd6b9.png" width="500">
All metadata points to original author, this repository, and this community. You can <a href="https://www.nuget.org/packages/CLI11/">find the package here</a>.
Please let me know if that's ok. And if not, I will promptly delist the package.

std::tuple<std::string, std::string>{"thanks", "sorry"};

**Comment by henryiii** [comment]
I think in general we want to encourage these sorts of third party packaging. :) I'd like to have some way to list them (see #774) without expanding our readme infinitely; ideally a page in the gitbook would work (but I really need to rework the gitbook into some non-extinct static site generator).

**Reply by diogo-strube** [reply]
Listing distinct packages (for different package managers) is not something I noticed consistency across open-source projects. A few projects list packages in a dedicated section in the Readme (such [nlohmann json](https://github.com/nlohmann/json)), some throw badges at the top of the Readme (like [MSBuildStructuredLog](https://github.com/KirillOsenkov/MSBuildStructuredLog)), and others list everything in a GitHub Page or a GitHub Wiki. Also, many go with the approach of listing nothing at all (like [AWS C++ SDKs](https://github.com/aws/aws-sdk-cpp)), I guess the reasoning is that a developer would check the "portal" of their choice to find any available/desired package.

But I can relate to your concern about growing the Readme file, as it is already quite verbose. Maybe a GitHub Wiki could allow moving existing detailed descriptions out of the Readme, leaving reduced content in the Readme that links to the Wiki (which can grow without cluttering).

As for the Doxygen documentation currently on the GitHub Page, what I have done on a few Doxygen projects was to update the html templates with projects like [Doxygen Awesome](https://github.com/jothepro/doxygen-awesome-css). It brings a fresher/modern look on pair to a few of the trendy dev ecosystems.

As for my opinion on how to list the available packages, I noticed [conan](https://conan.io/center/cli11) already appears as a badge at the top of the [Readme](https://github.com/CLIUtils/CLI11), so badges could continue to be used to point to any given package manager.
![image](https://user-images.githubusercontent.com/19294348/190871621-80435677-774f-40a4-92ba-91e9e54db370.png)

But back to this specific NuGet Pkg, I am wondering if I should create a PR to have the .nuspec file (describing the created CLI11 NuGet Pkg) committed to the CLI11 official repository, making sure the community can easily update any metadata as desired. I could give you all access to an Azure Pipeline you can go to click a Build button... or maybe add the build/deployment as a GitHub Action... let me know if that would be desired.
